### ISSUE TITLE
Lint/EmptyWhen false positive with inline comments.

### ISSUE THREAD

**Issue by marcandre** [rubocop/rubocop#7999]
0) Proper bug: the following will detect an offense even with `'AllowComments' => true`

```
case foo
when :bar then # do nothing
end
```

Note that the following is rightfully accepted
```
case foo
when :bar
  # do nothing
end
```

The spec file has multiple issues and is in need of love:
1) Explicit testing of `AllowComments` is minimal, should be expanded
2) Doesn't use `except_offense`/`no_offense` so spec failures are next to impossible to locate.
3) The default is `AllowComment: true`, but the `:config` spec will not load the defaults unless given there is a `cop_config` method. Add `let(:cop_config) { {} }` to the current spec and it fails (related to #7970)

**Comment by Darhazer** [comment]
Should 
```
case foo
when :bar # do nothing
end
```
also be accepted?
